daemon/src/setup_contract_actor.rs

```diff
@@ -0,0 +1,310 @@
+            &own_cets,
+            &msg1.cets,
+            &commit_desc,
+            commit_amount,
+        )
+        .unwrap();
+
+        let lock_tx = own_cfd_txs.lock;
+        let refund_tx = own_cfd_txs.refund.0;
+
+        verify_signature(
+            &refund_tx,
+            &commit_desc,
+            commit_amount,
+            &msg1.refund,
+            &params.other.identity_pk,
+        )
+        .unwrap();
+
+        let mut cet_by_id = own_cets
+            .into_iter()
+            .map(|(tx, _, msg, _)| (tx.txid(), (tx, msg)))
+            .collect::<HashMap<_, _>>();
+
+        FinalizedCfd {
+            identity: sk,
+            revocation: rev_sk,
+            publish: publish_sk,
+            lock: lock_tx,
+            commit: (commit_tx, *msg1.commit),
+            cets: msg1
+                .cets
+                .into_iter()
+                .map(|(txid, sig)| {
+                    let (cet, msg) = cet_by_id.remove(&txid).expect("unknown CET");
+
+                    (cet, *sig, msg)
+                })
+                .collect::<Vec<_>>(),
+            refund: (refund_tx, msg1.refund),
+        }
+    };
+
+    (actor, sender)
+}
+
+#[derive(Clone)]
+#[allow(dead_code)]
+pub enum OwnParams {
+    Maker(PartyParams),
+    Taker(PartyParams),
+}
+
+/// Role of the actor's owner in the upcoming contract
+enum Role {
+    Maker,
+    Taker,
+}
+
+/// A convenience struct for storing PartyParams and PunishParams of both
+/// parties and the role of the caller.
+struct AllParams {
+    pub own: PartyParams,
+    pub own_punish: PunishParams,
+    pub other: PartyParams,
+    pub other_punish: PunishParams,
+    pub own_role: Role,
+}
+
+impl AllParams {
+    fn new(
+        own: PartyParams,
+        own_punish: PunishParams,
+        other: PartyParams,
+        other_punish: PunishParams,
+        own_role: Role,
+    ) -> Self {
+        Self {
+            own,
+            own_punish,
+            other,
+            other_punish,
+            own_role,
+        }
+    }
+
+    fn maker(&self) -> &PartyParams {
+        match self.own_role {
+            Role::Maker => &self.own,
+            Role::Taker => &self.other,
+        }
+    }
+
+    fn taker(&self) -> &PartyParams {
+        match self.own_role {
+            Role::Maker => &self.other,
+            Role::Taker => &self.own,
+        }
+    }
+
+    fn maker_punish(&self) -> &PunishParams {
+        match self.own_role {
+            Role::Maker => &self.own_punish,
+            Role::Taker => &self.other_punish,
+        }
+    }
+    fn taker_punish(&self) -> &PunishParams {
+        match self.own_role {
+            Role::Maker => &self.other_punish,
+            Role::Taker => &self.own_punish,
+        }
+    }
+}
+
+fn verify_cets(
+    oracle_pk: &schnorrsig::PublicKey,
+    maker: &PartyParams,
+    taker_cets: &[(
+        Transaction,
+        EcdsaAdaptorSignature,
+        Vec<u8>,
+        schnorrsig::PublicKey,
+    )],
```

Comment:
Nit: since this function is called by both parties now, it's not `maker: PartyParams` and `taker_cets` but `other: PartyParams` and `own_cets...`
Same goes for the usage of `maker_encsig` in the function body.

Reply:
indeed, I missed that one - fixed!